app/javascript/components/widgets/weather/weather-effect.jsx

```diff
@@ -0,0 +1,386 @@
+  height: 4px;
+  border-radius: 4px;
+  animation: ${SnowDropFrames} linear, ${HeavySnowFallFrames} ease-in-out;
+`;
+
+const SnowFall = styled(Drop)`
+  background: white;
+  width: 3px;
+  height: 3px;
+  border-radius: 3px;
+  animation: ${SnowDropFrames} linear, ${SnowWaveFrames} ease-in-out;
+`;
+
+const SnowContainer = styled.div`
+  width: 100%;
+  height: 100%;
+  position: absolute;
+`;
+
+const snowStates = {
+  // Use IDs, pass snow type, drop variant, array of extra types
+  600: [SnowFall, 6, 2, 20], // Light
+  601: [SnowFall, 4, 2, 50], // Normal
+  602: [SnowFall, 2, 2, 70], // Heavy
+  611: [HeavySnowFall, 3, 2, 20], // Sleet
+  612: [HeavySnowFall, 2, 1, 20], // Sleet
+  613: [HeavySnowFall, 1.5, 1, 30], // Sleet
+  615: [SnowFall, 4, 2, 20, 0], // Light rain and snow
+  616: [SnowFall, 2, 2, 20, 1], // Rain and snow
+  620: [SnowFall, 6, 1, 20], // Light
+  621: [SnowFall, 4, 1, 50], // Normal
+  622: [SnowFall, 2, 1, 70], // Heavy
+};
+
+const Snow = ({ id }) => {
+  const state = snowStates[id];
+  if (state === undefined) return null;
+  const [Type, duration, variance, count, rain] = state;
+  return (
+    <>
+      <SnowContainer>
+        <WeatherWrapper {...{ Type, duration, variance, count }} />
+      </SnowContainer>
+      {rain !== undefined ? <Rain id={rain} /> : null}
+    </>
+  );
+};
+
+Snow.propTypes = {
+  id: PropTypes.number.isRequired,
+};
+
+// #endregion
+// #region Thunder
+
+// Reference: https://codepen.io/Chrislion_me/pen/rVqwbO
+const LightningKeyframes = keyframes`
+{
+  0% {opacity: 0;}
+
+  40% {opacity: 0%;}
+  41% {opacity: 50%;}
+  43% {opacity: 20%;}
+  44% {opacity: 90%;}
+  50% {opacity: 0%;}
+
+  100% {opacity: 0%;}
+}
+`;
+
+const ThunderClouds = styled.div`
+  position: absolute;
+  width: 100%;
+  height: 100%;
+  background: #dfff;
+  animation: ${LightningKeyframes} ease-out 8s infinite;
+`;
+
+const Clouds = styled.div`
+  position: absolute;
+  width: 100%;
+  height: 100%;
+  background: linear-gradient(to bottom, #999, #9999 40%, #9992 80%, #000b);
+`;
+
+const thunderStates = {
+  200: 1,
+  201: 2,
+  202: 3,
+  230: 0,
+  231: 0,
+  232: 0,
+};
+
+const Thunder = ({ id }) => {
+  const rain = thunderStates[id];
+
+  return (
+    <>
+      {rain !== undefined ? <Rain id={rain} /> : null}
+      <Clouds />
+      <ThunderClouds />
+    </>
+  );
+};
+
+Thunder.propTypes = {
+  id: PropTypes.number.isRequired,
+};
+
+// #endregion
+
+const effectMap = {
+  Drizzle,
+  Rain,
+  Snow,
+  Thunder,
+};
+
+const WeatherEffect = ({ weather }) => {
+  const {
+    current: {
+      weather: { main, id },
+    },
+  } = weather;
+
+  const [mainText, setMain] = useState(main);
+  const [idText, setId] = useState(id.toString());
+
+  useEffect(() => {
+    setId(id);
+    setMain(main);
+  }, [id, main]);
+
+  const Effect = effectMap[mainText];
+
+  return (
+    <>
+      {Effect ? (
+        <EffectContainer>
+          <Effect id={parseInt(idText !== '' ? idText : 0, 10)} />
```

Comment:
> it also locks the input text to always be a valid number (such as 0)

What do you mean? I think it can still be undefined? or anything else really, this is javascript there are no rules.

```js
  const [id, setId] = useState(value);
  const Effect = effectMap[mainText];

  return (
    <>
      {Effect && id ? (
         <EffectContainer>
           <Effect id={id}/>
         </EffectContainer>
        ):
 ....
        
      
 ```